pkg/registry/core/pod/storage/storage.go

```diff
@@ -256,6 +256,11 @@ func (r *BindingREST) setPodNodeAndMetadata(ctx context.Context, podUID types.UI
 			Type:   api.PodScheduled,
 			Status: api.ConditionTrue,
 		})
+		// Clear nomination hint to prevent stale information affecting external components.
+		if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.NominatedNodeNameForExpectation) {
```

Comment:
not having read the KEP before reviewing this, this is a surprising behavior to associate with a feature named "NominatedNodeNameForExpectation" ... clearing the NominatedNode field after binding (and not letting it be set on an already-bound pod) makes a lot of sense... is there a clearer name we could use for the gate of this behavior change?

Reply:
I don't feel that strongly, it's just pretty non-obvious why this field behavior is tied to that gate, and if someone has to disable the gate in order to mitigate some unexpected scheduler behavior, it might surprise them to also lose the API server field-clearing behavior if that wasn't actually problematic.

Reply:
https://github.com/kubernetes/kubernetes/pull/132994